src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/IProjectHotReloadSession.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.HotReload
+{
+    public interface IProjectHotReloadSession
+    {
+        bool IsSupported { get; }
+
+        Task StartSessionAsync(CancellationToken cancellationToken);
+
+        Task StopSessionAsync(CancellationToken cancellationToken);
+
+        Task ApplyChangesAsync(CancellationToken cancellationToken);
+
+        Task<bool> ApplyLaunchVariablesAsync(IDictionary<string, string> envVars, string configuration, CancellationToken cancellationToken);
```

Comment:
@BillHiebert What is `configuration` needed for?

Reply:
@tmeschter it is only supported for debug builds, so the idea was that the caller could just blindly call this and have it return false if not supported.  We could remove it and delegate that responsibility to the caller

Reply:
One concern I have is that we shouldn't rely on the configuration _name_ in this check (whether it is done here or in the caller). Do we know what settings in Hot Reload depend son in Debug configurations?

Reply:
FYI, it looks like we really need to check for the following:

``` xml
<Optimize>false</Optimize>
<EmitDebugInformation>true</EmitDebugInformation>
```